File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0 , buy = prices[0] , sell = prices[0] ;
        int n = prices.length ;
        for(int price : prices){
            if( buy > price ){
                profit += sell-buy;
                buy = price;
                sell = price;
            }
            if(sell<price && sell<=buy){
                sell = price;
                profit += sell - buy ;
                buy = price ;
            }
        } 
        return profit ;
    }
}

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        buy = prices[0]
        sell = prices[0]
        profit = sell - buy
        maxi = 0

        for i in prices :
            if i < buy or sell > i :
                buy = i
                sell = i
                
            if sell<i :
                sell = i
                profit = sell-buy
                maxi += profit
                buy = i 
            
        return maxi
